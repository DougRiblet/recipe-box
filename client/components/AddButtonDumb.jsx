import React from 'react'

const AddButtonDumb = ({ addRecipe }) => (
  <button
    id='addRecipeButton'
    onClick={() => addRecipe()}
  >Add New Recipe</button>
)

AddButtonDumb.propTypes = {
  addRecipe: React.PropTypes.func.isRequired
}

export default AddButtonDumb
