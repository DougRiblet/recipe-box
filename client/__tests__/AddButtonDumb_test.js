import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AddButtonDumb from '../components/AddButtonDumb'

describe('AddButtonDumb', () => {
  test('Snapshot', () => {
    const AddButtonDumb_tree = renderer.create(
      <AddButtonDumb
        addRecipe={() => addRecipe()}
      />
    ).toJSON()
    expect(AddButtonDumb_tree).toMatchSnapshot()
  })

  test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AddButtonDumb
      addRecipe={() => addRecipe()}
    />, div)
  })
})
