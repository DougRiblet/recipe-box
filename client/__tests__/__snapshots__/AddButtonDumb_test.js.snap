// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButtonDumb Snapshot 1`] = `
<div
  id="add-recipe-div"
>
  <button
    id="addRecipeButton"
    onClick={[Function]}
  >
    Add New Recipe
  </button>
</div>
`;
